src/Controls/src/Core/Xaml/TypeConversionExtensions.cs

```diff
@@ -42,24 +43,24 @@ static class TypeConversionExtensions
 			{ typeof(Font), typeof(FontTypeConverter) }
 		};
 
-		private static Dictionary<Type, Func<TypeConverter>> KnownConverterFactories = new()
+		static Dictionary<Type, Func<TypeConverter>> KnownConverterFactories = new()
 		{
 			{ typeof(Font), () => new FontTypeConverter() }
```

Comment:
General question for @StephaneDelcroix, should we have a lot more converter types in this dictionary? I'm thinking about types like `Color`, `Thickness`, etc.

Reply:
@jonathanpeppers glad you asked

well, at least we shouldn't have multiple caches... TypeConversionExtensions.KnownConverters, TCE.KnownConverterFactories, BindableProperty.KnownTypeConverter; and then we have NodeILExtensions.KnonwnCompiledTypeConverters

Type Conversion happens at multiple places:
- when setting a value from xaml
- when applying a binding (this includes AppThemeBinding, DynamicResource, ...)
- when applying a style

ColorTypeConverter and ThicknessTypeConverter do _not_ need to be cached in a known list, as they are attributed with a TypeConverterAttribute, they need to be in the KnownCOMPILEDTypeConverter as they're not attributed with the ProvideCompiledAttibute.

The only converter that can not be retrieved using reflection is UriTypeConverter as System.Uri is NOT attributed on .NET core 6 (even though System provides a UriTypeConverter that we should use now that we replaced our own base TypeConverter type by System.ComponentModel.TypeConverter. Side note, System.Uri is attributed in the net framework profile, go figure). so why do we whitelist ColorTypeConverter and EasingTypeConverter in BindableProperty.cs is something that need clarification.

I don't recall the details of why FontTypeConverter need to be whitelisted, but I recall that we almost got rid of Font as a property type in most (all?) of our controls (because compound properties are a mess and produce too many propertychange events, and redraw).

We could/should cache the TypeConverters:
- at runtime when we encounter them
- at startup time using reflection (fuck no)
- or at compile time using a sourcegen
Our own TypeConverters are known beforehand, but I'm not keen on maintaining a list of those (I'd prefer a sourcegen), but there's also the TypeConverters defined in 3rd party assemblies. Oh, and caching type converters isn't a solution for avoiding reflection, as TypeConverterAttributes on properties take priority over the ones defined on Type, but we could cache that too (using the PropertyInfo as key e.g.)

To summarise this digression on TypeConverters, here's a list of action points:
- [ ] Investigate the usage of `Font` in Controls. It's probably unused, in that case `FontTypeConverter` and the 2 caches in TypeConversionExtension could go away
- [ ] Investigate `Font` usage in Core, and the reason it was ported there, and not deprecated
- [ ] Investigate why ColorTypeConverter and EasingTypeConverter are whitelisted
- [ ] Investigate caching TypeConverters for properties and types. I'd recommend going for a sourcegen solution and generate a switch instead of a dictionary to avoid allocation bump
- [ ] None of our TypeConverters have any side effect, and instances could be reused (we can speak for 3rd party converters, unfortunately...)

Reply:
We need to track the progress of this in a different issue...